### PR TITLE
fix: update visibility for elements with textContent but without width/height

### LINKED ISSUE #29687: Parent width should not signal element hidden when it's visible
### Current behavior

Parent element width: 0 on element visible in browser will be detected as hidden in Cypress. This is not correct. 

### Desired behavior

This check should not be done on parent when not needed. Only on element.
Element visible in browser should be clickable and not marked as hidden when they visible

### Test code to reproduce

```html
<div id="id2">
  <div style="width: 0">
    <p class="big">IN SECTION 2 </p>
  </div>
</div>
```

```js
expect(cy.$$('#id2 >div > p')).to.be.visible
    
cy.get('#id2 >div > p').click()
```

### Cypress Version

13.11.0

### Node version

v22.3.0

### Operating System

Windows 11

### Debug Logs

_No response_

### Other

I already work on fix and PR.

### PR DESCRIPTION
**Pull request by senpl**
- Closes https://github.com/cypress-io/cypress/issues/29687

### Additional details
* Fixed visibility issue when the element has no width/height but does have `textContent`.

### Steps to test
See new tests for scenarios.

### How has the user experience changed?
Before:
Screenshot 2024-11-26 at 2 10 04 PM

After:
Screenshot 2024-11-26 at 2 09 30 PM

### PR Tasks

- [x] Have tests been added/updated?
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by senpl** [comment]
It was done similar in the past as could be seen in https://github.com/cypress-io/cypress/issues/5974

**Comment by senpl** [comment]
One test updated because element that was in test in fact was visible, and incorrectly exected to be hidden and not visible.

**Review by jennifer-shehane — DISMISSED** [review]
@senpl There a failure regarding visibility in the tests.
Screenshot 2024-06-21 at 1 38 47 PM

**Comment by senpl** [comment]
Now looks like cert errors, so probably something with pipeline.

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -46,6 +47,7 @@ _Released 6/18/2024_
 - Fixed an issue where `inlineSourceMaps` was still being used when `sourceMaps` was provided in a users typescript config for typescript version 5. Fixes [#26203](https://github.com/cypress-io/cypress/issues/26203).
 - When capture protocol script fails verification, an appropriate error is now displayed. Previously, an error regarding Test Replay archive location was shown. Addressed in [#29603](https://github.com/cypress-io/cypress/pull/29603).
 - Fixed an issue where receiving HTTP responses with invalid headers raised an error. Now cypress removes the invalid headers and gives a warning in the console with debug mode on. Fixes [#28865](https://github.com/cypress-io/cypress/issues/28865).
+- Fixed an issue where parent width signal element hidden when it's visible [#29687](https://github.com/cypress-io/cypress/issues/29687).
```
Comment (MikeMcC399) [inline cli/CHANGELOG.md]:
```suggestion
```

This should be removed

File: packages/driver/cypress/e2e/commands/actions/click.cy.js [hunk 2]
```diff
@@ -2205,7 +2205,7 @@ describe('src/cy/commands/actions/click', () => {
 
           expect(logsArr).to.have.length(4)
           expect(lastLog.get('error')).to.eq(err)
-          expect(err.message).to.include('`cy.click()` failed because this element is not visible')
+          expect(err.message).to.include('`cy.click()` failed because this element')
```
Comment (MikeMcC399) [inline packages/driver/cypress/e2e/commands/actions/click.cy.js:2208]: Does this error message make sense now?

**Comment by jennifer-shehane** [comment]
We've decided to prioritize looking into these visibility fixes in Quarter 4 of this year, when we intend to spend time on Cypress 14 release. It will ease any concerns we have around releasing breaking changes with these fixes and also help us plan the proper time to review these. Tracking here: https://github.com/cypress-io/cypress/issues/29925

**Comment by jennifer-shehane** [comment]
Updated the base branch to go against release/14.0.0

**Comment by jennifer-shehane** [comment]
I'll leave some notes here on where I left off on this. 

Essentially, there are some circumstances where elements should be considered visible/hidden depending on the state of their child node that is a TEXT node. And Cypress is not taking those situations into account today. 

- If an element has a child text node that contains characters other than whitespace characters, that content makes it visible (even if there is no width/height of the element), unless the element itself is explicitly hidden OR **the text node is hidden by the element, like the element having overflow: hidden and the text node being clipped**. 

^ this last part in highlighted text is not currently being accounted for in the logic, which is why there is a failing test. I'm not sure if there are other circumstances that this updated code doesn't account for. Essentially the Cypress logic never seemed to be taking into account TEXT nodes, we've only been accounting for actual element nodes.

**Review by AtofStryker — APPROVED** [review]

**Pull request merged**